crates/re_renderer/src/context.rs

```diff
@@ -399,6 +394,40 @@ impl RenderContext {
                 .push(self.queue.submit([command_buffer]));
         }
     }
+
+    /// Gets a renderer with the specified type, initializing it if necessary.
+    pub fn get_renderer<R: 'static + Renderer + Send + Sync>(
+        &self,
+    ) -> MappedRwLockReadGuard<'_, R> {
+        {
+            // Most likely we already have the renderer. Take a read lock and return it.
+            if let Ok(renderer) =
+                parking_lot::RwLockReadGuard::try_map(self.renderers.read(), |r| r.get::<R>())
+            {
+                return renderer;
+            }
+        }
+
+        // If it wasn't there we have to add it.
+        // This path is rare since it happens only once per renderer type in the lifetime of the ctx.
+        // (we don't discard renderers ever)
+        {
+            self.renderers.write().get_or_create::<_, R>(
+                &self.shared_renderer_data,
+                &self.gpu_resources,
+                &self.device,
+                &self.resolver,
+            );
+        }
+        // Release write lock again and only take a read lock.
+        // safe to unwrap since we just created it.
+        parking_lot::RwLockReadGuard::map(self.renderers.read(), |r| r.get::<R>().unwrap())
```

Comment:
We shouldn't need `unwrap`. `get_or_create` should just return the gotten or created thing

```suggestion
        self.renderers.write().get_or_create::<_, R>(
            &self.shared_renderer_data,
            &self.gpu_resources,
            &self.device,
            &self.resolver,
        )
```

Reply:
the problem with that is that we have to release the write lock first! The result of `get_or_create` implies keeping the write lock and we don't want to do that.